### ISSUE TITLE
Clarify intended behaviour of `reportAssertAlwaysTrue`

### ISSUE THREAD

**Issue by tpgillam** [microsoft/pyright#9050]
I just discovered this feature in the documentation, described as flagging an 'assert statement that will provably always assert'. Given the discussion in #9046, we don't want a rule with this behaviour.

And indeed, none of the following statements trigger it despite pyright being able to prove their truth in other contexts:

```python
assert True
assert 1 == 1
assert None is None
assert 1 in (1, 2)
assert min
```

It only flags the case of implicit `true`-ness of a non-empty tuple:
```python
assert (1,)       # error
assert (1, 2)     # error
assert (1, 2, 3)  # error
```
which happens [here](https://github.com/microsoft/pyright/blob/ed4d23e784f74fa88314325250cc9bb6bec0e9b6/packages/pyright-internal/src/analyzer/checker.ts#L1129-L1143)

Perhaps the scope of this rule can be narrowed in the documentation? A more specific name might be nice too.

**Comment by erictraut** [comment]
This check is specifically meant to catch the case where a single parenthesized tuple expression is used instead of a condition expression separated from an error string. This is a common footgun in Python.

```python
def func1(v: int):
    # This is was intended
    assert v > 0, "Bad value"

def func2(v: int):
    # This is not intended
    assert (v > 0, "Bad value")
```

I've updated the documentation to make the intent clearer for this check.

This check is not intended to report any `assert` whose condition statically evaluates to true. That would produce many noisy false positives in well-written typed Python code bases. There are many legitimate uses for asserts whose condition expression appears (from a static analysis standpoint) to always evaluate to true or false. This isn't generally indicative of a programming error.
